### ISSUE TITLE
Debug assertion failure: AlongStep.hh:110: `speed > 0`

### ISSUE THREAD

**Issue by amandalund** [celeritas-project/celeritas#616]
#### Failure:
```json
    "failure": {
     "condition": "speed > 0",
     "file": "/home/alund/celeritas_project/celeritas/src/celeritas/global/alongstep/AlongStep.hh",
     "line": 110, 
     "type": "DebugError",
     "which": "internal assertion failed"
    }  
```
#### Regression problem:
cms2018-vecgeom-cpu
<details><summary>JSON input</summary><pre>
{
"_geometry": "vecgeom",
"_instance": 0,
"_name": [
"cms2018",
"vecgeom",
"cpu"
],
"_num_events": 7,
"_num_primaries": 9100,
"_outdir": "cms2018-vecgeom-cpu",
"_timeout": 600.0,
"brem_combined": false,
"eloss_fluctuation": true,
"enable_diagnostics": false,
"geant_options": {
"coulomb_scattering": false,
"eloss_fluctuation": false,
"em_bins_per_decade": 56, 
"lpm": true,
"msc": "none",
"physics": "em_basic",
"rayleigh_scattering": true
},
"geometry_filename": "/home/alund/celeritas_project/regression/input/cms2018.gdml",
"hepmc3_filename": "/home/alund/celeritas_project/regression/input/simple-cms-13TeV.hepmc3",
"initializer_capacity": 1048576,
"mag_field": [
0.0,
0.0,
0.0
],
"max_events": 7,
"max_num_tracks": 4096,
"max_steps": 524288,
"physics_filename": "/home/alund/celeritas_project/regression/input/cms2018.gdml",
"secondary_stack_factor": 3.0,
"seed": 20220904,
"sync": true,
"use_device": false
}
</pre></details>

#### Diagnosis:
The electron has a pre-step energy = 5.83609639577594e-03. The step is range-limited, with range = 1.00002899990615401e-01. In the range-to-step conversion, the range is slightly larger than the min range x fudge factor, so the step is scaled to step = 1.00002899923337801e-01. The mean energy loss = 5.83609639577594309e-03, resulting in a super small but nonzero post-step energy = 1.73472347597681e-18. Because the energy is so low, the calculated beta^2 is zero due to rounding error, and the speed is zero.

**Comment by sethrj** [comment]
This suggests we need a speed cutoff for "effectively stopped": 1e-18 MeV is unrealistically close to zero so we need some sort of hard cutoff (was that only done for MSC?), and maybe ensure that it's defined so that `cutoff_energy**2 > 0`

**Comment by sethrj** [comment]
Failures in `cms2018-vecgeom-cpu`:

```json
"failure": {
 "condition": "speed > 0",
 "context": {
  "dir": [
   0.984471131266238,
   0.14771995186801973,
   -0.09484412224004088
  ],
  "energy": [
   1.734723475976807e-18,
   "MeV"
  ],
  "event": 4,
  "label": "along-step-general-linear",
  "num_steps": 1,
  "parent": 393,
  "particle": 1,
  "pos": [
   38.5587122234001,
   7.806680082261858,
   227.28506044268605
  ],
  "thread": 1653,
  "track": 28317,
  "type": "KernelContextException",
  "volume": 2126
 },
 "file": "/home/s3j/.local/src/celeritas/src/celeritas/global/alongstep/AlongStep.hh",
 "line": 110,
 "type": "DebugError",
 "which": "internal assertion failed"
}
```
and
```json
"failure": {
 "condition": "speed > 0",
 "context": {
  "dir": [
   -0.7764297365878872,
   0.6090481744190231,
   0.1619172176728581
  ],
  "energy": [
   9.540979117872439e-18,
   "MeV"
  ],
  "event": 6,
  "label": "along-step-general-linear",
  "num_steps": 1,
  "parent": 249599,
  "particle": 1,
  "pos": [
   103.5844707518932,
   36.381341521654974,
   310.5217398600153
  ],
  "thread": 408,
  "track": 672611,
  "type": "KernelContextException",
  "volume": 3413
 },
 "file": "/home/s3j/.local/src/celeritas/src/celeritas/global/alongstep/AlongStep.hh",
 "line": 110,
 "type": "DebugError",
 "which": "internal assertion failed"
}
```
